gulp-tasks/publish-bundle.js

```diff
@@ -0,0 +1,211 @@
+const gulp = require('gulp');
+const GitHubApi = require('github');
+const path = require('path');
+const semver = require('semver');
+const glob = require('glob');
+const fs = require('fs-extra');
+const TarGz = require('tar.gz');
+
+const logHelper = require('./utils/log-helper');
+const constants = require('./utils/constants');
+const spawnPromiseWrapper = require('./utils/spawn-promise-wrapper');
+const uploadBundleToCDN = require('./utils/upload-bundle-to-cdn');
+
+const GITHUB_OWNER = 'GoogleChrome';
+const GITHUB_REPO = 'workbox';
+
+const BUILD_PATH = path.join(__dirname, '..', 'github-releases');
+
+const github = new GitHubApi();
+
+github.authenticate({
+  type: 'token',
+  token: process.env.GITHUB_TOKEN,
+});
+
+const downloadGitCommit = (tagName) => {
+  logHelper.log(`Clone Git repo for tag: '${tagName}'.`);
+  const tempPath = path.join(BUILD_PATH, tagName, 'source-code');
+  const params = [
+    'clone',
+    '--branch', tagName,
+    '--depth', '1',
+    `http://github.com/${GITHUB_OWNER}/${GITHUB_REPO}.git`,
+    tempPath,
+  ];
+
+  return spawnPromiseWrapper('git', params)
+  .then(() => tempPath);
+};
+
+const buildProject = (projectPath) => {
```

Comment:
This is running the build task in the context of what ever the tagged code is (i.e. any missing tags are pulled down, installed their dependencies and running gulp in that project).

I.e. this doesn't necessarily mean we are going to have the same build process.

Happy to change approach if you think it would be better (i.e. a node script that handles the github and CDN publishing seperately from our build process?)